plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -324,6 +328,13 @@ private Stream<DeltaLakeTransactionLogEntry> getV2CheckpointTransactionLogEntrie
                 });
     }
 
+    private static Set<CheckpointEntryIterator.EntryType> filterDataTypes(Set<CheckpointEntryIterator.EntryType> entryTypes)
+    {
+        return entryTypes.stream()
+                .filter(entryType -> entryType == ADD || entryType == REMOVE)
+                .collect(toImmutableSet());
```

Comment:
nit: We could use [Sets.intersection](https://guava.dev/releases/snapshot/api/docs/com/google/common/collect/Sets.html#intersection(java.util.Set,java.util.Set)):
```java
return Sets.intersection(entryTypes, Set.of(ADD, REMOVE));
```